src/main/java/com/jpr/maintenance/database/repository/EntityTemplateUserRepositoryImpl.java

```diff
@@ -0,0 +1,26 @@
+package com.jpr.maintenance.database.repository;
+
+import com.jpr.maintenance.database.model.UserEntity;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.r2dbc.core.R2dbcEntityTemplate;
+import reactor.core.publisher.Mono;
+
+import static com.jpr.maintenance.reflection.ObjectMapper.toObject;
+
+@RequiredArgsConstructor
+public class EntityTemplateUserRepositoryImpl implements EntityTemplateUserRepository {
+
+    private final R2dbcEntityTemplate template;
+
+    @Override
+    public Mono<UserEntity> saveUser(UserEntity userEntity) {
+        return template.getDatabaseClient()
+            .sql("INSERT INTO \"user\" (username, \"password\", salt) VALUES ($1, $2, $3) RETURNING id,username,\"password\",salt")
```

Comment:
Shouldn't we also bind the id here? If I understand the code correctly, like this we still will not be returning the primary key.